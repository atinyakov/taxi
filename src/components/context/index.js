import { createContext } from 'react'

export const status = {
    loggedIn: false
};

// export const login = () => {

// }

export const isLoggedIn = createContext(
    status.loggedIn // значеине по умолчанию
);

export const login = createContext(
    {handleLogin : () => {
        console.log('here');
    }}
);