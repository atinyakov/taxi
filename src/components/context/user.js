import { createContext } from 'react'

export const userData = {
    isLoggedIn: false,
    username: '',
    password: ''
}

export const reducer = (state, action) => {
    switch (action.type) {
      case 'login':
          console.log('here')
        return {
            ...state,
            isLoggedIn: true
        };
      case 'logout':
        return {
            ...state,
            isLoggedIn: false
        };
      default:
        return state
    }
  }

export const user = createContext(
    userData
);