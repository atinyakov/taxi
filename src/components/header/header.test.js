import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { appContext, userContext } from '../context';
describe('Header', () => {
    it("matches snapshot", () => {
        const { asFragment } = render(
            <appContext.Provider value={{ toggleLogginPopup: ()=>{}, toggleMap: ()=>{}, toggleProfile: ()=>{} }}>
                <userContext.Provider value={{ isLoggedin: true }}>
                    <Header />
                </userContext.Provider>
            </appContext.Provider>);
        expect(asFragment()).toMatchSnapshot();
    });
});



