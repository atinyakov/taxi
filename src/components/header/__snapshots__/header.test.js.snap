// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
      data-testid="header"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-1 MuiTypography-h6"
        >
          Loft Taxi
        </h6>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Карта
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Профиль
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            Выйти
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</DocumentFragment>
`;
